api/tokens.go

```diff
@@ -124,52 +123,35 @@ func (api *API) TokensHandler(ctx context.Context) http.HandlerFunc {
 }
 
 //SignOutHandler invalidates a users access token signing them out and returns a http handler interface
-func (api *API) SignOutHandler(ctx context.Context) http.HandlerFunc {
-	return func(w http.ResponseWriter, req *http.Request) {
-		var errorList []apierrorsdeprecated.Error
-
-		authString := req.Header.Get(AccessTokenHeaderName)
-		if authString == "" {
-			invalidTokenErrorBody := apierrorsdeprecated.IndividualErrorBuilder(apierrorsdeprecated.InvalidTokenError,
-				apierrorsdeprecated.MissingTokenDescription)
-			errorList = append(errorList, invalidTokenErrorBody)
-			log.Event(ctx, "no authorization header provided", log.ERROR)
-			errorResponseBody := apierrorsdeprecated.ErrorResponseBodyBuilder(errorList)
-			apierrorsdeprecated.WriteErrorResponse(ctx, w, http.StatusBadRequest, errorResponseBody)
-			return
-		}
+func (api *API) SignOutHandler(w http.ResponseWriter, req *http.Request, ctx context.Context, errorResponse *models.ErrorResponse) {
+	accessToken := models.AccessToken{
+		AuthHeader: req.Header.Get(AccessTokenHeaderName),
+	}
+	validationErr := accessToken.Validate(ctx)
+	if validationErr != nil {
+		errorResponse.Errors = append(errorResponse.Errors, validationErr)
+	}
 
-		authComponents := strings.Split(authString, " ")
-		if len(authComponents) != 2 {
-			log.Event(ctx, "malformed authorization header provided", log.ERROR)
-			invalidTokenErrorBody := apierrorsdeprecated.IndividualErrorBuilder(apierrorsdeprecated.InvalidTokenError,
-				apierrorsdeprecated.MalformedTokenDescription)
-			errorList = append(errorList, invalidTokenErrorBody)
-			errorResponseBody := apierrorsdeprecated.ErrorResponseBodyBuilder(errorList)
-			apierrorsdeprecated.WriteErrorResponse(ctx, w, http.StatusBadRequest, errorResponseBody)
-			return
-		}
+	if len(errorResponse.Errors) > 0 {
+		errorResponse.Status = http.StatusBadRequest
+		return
+	}
 
-		_, err := api.CognitoClient.GlobalSignOut(
-			&cognitoidentityprovider.GlobalSignOutInput{
-				AccessToken: &authComponents[1]})
+	_, err := api.CognitoClient.GlobalSignOut(accessToken.GenerateSignOutRequest())
 
-		if err != nil {
-			log.Event(ctx, "From Cognito - "+err.Error(), log.ERROR)
-			invalidTokenErrorBody := apierrorsdeprecated.IndividualErrorBuilder(err, "")
-			errorList = append(errorList, invalidTokenErrorBody)
-			errorResponseBody := apierrorsdeprecated.ErrorResponseBodyBuilder(errorList)
-			isInternalError := apierrorsdeprecated.IdentifyInternalError(err)
-			if isInternalError {
-				apierrorsdeprecated.WriteErrorResponse(ctx, w, http.StatusInternalServerError, errorResponseBody)
-			} else {
-				apierrorsdeprecated.WriteErrorResponse(ctx, w, http.StatusBadRequest, errorResponseBody)
-			}
-			return
+	if err != nil {
+		responseErr := models.NewCognitoError(ctx, err, "Cognito GlobalSignOut request for signout")
+		errorResponse.Errors = append(errorResponse.Errors, &responseErr)
```

Comment:
If we make the change mentioned above, not passing in an errorResponse struct to this handler, then you don't need to append here you can just instantiate. The benefit of this is it gives less impression that there might already be a few errors waiting to be processed; we're more clearly saying "This function I just called might have a couple of errors, but I know I've handled all errors before this point"

Reply:
Changed to return a new ErrorResponse when an error is created.